R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+    {
+        return("Error: bin_method argument must be one of 'quantile', 'pretty', 'equal'")
+    }
+
+    # Verify bins
+    if(is.null(bins) || class(bins) != "numeric")
+    {
+        return("Error: bins argument must be a positive integer and not NULL")
+    }
+    if(bins < 1 || bins > 100)
+    {
+        return("Error: bins argument must be a valid integer between 1-100")
+    }
+
+    # verify dpi
+    if(!"numeric" %in% (class(dpi)) || dpi < 30 || dpi > 300)
+    {
+        return("Error: DPI argument must be a numeric value between 30 and 300")
+    }
+
+    # Verify expand_xy
+    if(!"numeric" %in% (class(expand_xy)) || length(expand_xy) != 2)
+    {
+        return("Error: expand_xy argument must be a numeric vector of length 2")
+    }
+
+    # Verify map_xy_min_max
+    if(!"numeric" %in% (class(map_xy_min_max)) || length(map_xy_min_max) != 4)
+    {
+        return("Error: map_xy_min_max argument must be a numeric vector of length 4")
+    }
+
+    # Verify title
+    if(!"character" %in% class(map_title) && !is.null(map_title))
+    {
+        return("Error: map_title must be of class character")
+    }
+
+    # Verify map_palette
+    if(!is.null(map_palette) && (!"character" %in% class(map_palette) || !map_palette %in% row.names(brewer.pal.info)) )
+    {
+        return("Error: map_palette must be of class character and a valid entry in the RColorBrewer palette (see brewer.pal.info")
+    }
+
+    # Verify palette_reverse
+    if(!"logical" %in% class(map_palette_reverse))
+    {
+        return("Error: map_palette_reverse must be of type logical (TRUE or FALSE)")
+    }
+
+    # Verify map_width_height_in
+    if(!"numeric" %in% (class(map_width_height_in)) || length(map_width_height_in) != 2)
+    {
+        return("Error: map_width_height_in argument must be a numeric vector of length 2")
+    }
+
+    # Verify map_legend_title
+    if(!"character" %in% class(map_legend_title) && !is.null(map_legend_title))
+    {
+        return("Error: map_legend_title must be of class character or NULL")
+    }
+
+    # Verify map_x_label
+    if(!"character" %in% class(map_x_label)  && !is.null(map_x_label))
+    {
+        return("Error: map_x_label must be of class character or NULL")
+    }
+
+    # Verify map_y_label
+    if(!"character" %in% class(map_y_label) && !is.null(map_y_label))
+    {
+        return("Error: map_y_label must be of class character or NULL")
+    }
+
+    return(output)
+}
+
+
+#' Return and output errors
+#'
+#' This function runs a check on csv data inputs and looks for errors
+#' @param error (Character) - Specific error string for output
+#' @param location (Character) - Location the error was caught
+#' @return (Character) - Prints the error to console and also returns it to caller
+#' @export
+return_error <- function(error, location)
+{
+    print(paste0("There was an error at - ", location, " - building your map:"))
```

Comment:
should you both print and return the error?  Will this be duplicative?

Reply:
It would be duplicative, but that was the idea. It would certainly be fine to just print the error string and return something empty like return() also. Or did you have an alternative idea?

Reply:
I believe a single error is fine.